src/test/services/user.test.ts

```diff
@@ -0,0 +1,246 @@
+import { USER_ROLES, UserRole } from "../../IsaacApiTypes";
+import {
+  extractTeacherName,
+  isAdmin,
+  isAdminOrEventManager,
+  isEventLeader,
+  isEventLeaderOrStaff,
+  isEventManager,
+  isLoggedIn,
+  isStaff,
+  isStudent,
+  isTeacher,
+  isTeacherOrAbove,
+  isTeacherPending,
+  isTutor,
+  isTutorOrAbove,
+  roleRequirements,
+  schoolNameWithPostcode,
+} from "../../app/services/user";
+import { School } from "../../IsaacAppTypes";
+import { mockUser } from "../../mocks/data";
+
+const generateTestCases = (functionName: string) => {
```

Comment:
Yeah it took me a while and I'm conscious it's less readable, but with over 100 test cases needed in this file (a lot of them very similar) I was trying to avoid writing out the test cases each time.   I did have an earlier version which was still using `generateTestCases` approach but giving a more explicit list of the scenarios (avoiding the .map and .push approach) which might be easier to read?

Reply:
I think it's fine as it is, don't worry. It's also harder to write tests for code that is not yours.